SourceX/DiabloUI/errorart.h

```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+namespace dvl {
+
+extern char btnData[56][111];
```

Comment:
1. These should be const.
2. Use an array literal insted of a C-string literal, these have no limit.

Should this be a `char` of particular signedness? Probably worth specifying explicitly.

Reply:
The downside to array literal is that it takes up a lot more text to represent, and this data is already pretty hefty, so I was hoping I could avoid that?

Reply:
If we use decimal instead of hex, it will likely be smaller than C-string, e.g.:

```c++
// clang-format off
const char btnData[6216] = {1,7,0,255};
```

Reply:
Well I'm a sucker for alignment, so will just accept it as an array literal, the compiled file is the same size in any case :)

Reply:
I can't get this to work with const. If I add it to both the header and the cpp file then GCC throws:
```
/usr/bin/ld: CMakeFiles/devilutionx.dir/SourceX/DiabloUI/button.cpp.o: in function `dvl::LoadSmlButtonArt()':
/home/ajenbo/code/diablo/devilutionX/SourceX/DiabloUI/button.cpp:12: undefined reference to `dvl::btnData'
/usr/bin/ld: CMakeFiles/devilutionx.dir/SourceX/DiabloUI/dialogs.cpp.o: in function `dvl::(anonymous namespace)::Init(char const*, char const*, bool)':
/home/ajenbo/code/diablo/devilutionX/SourceX/DiabloUI/dialogs.cpp:217: undefined reference to `dvl::popupData'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/devilutionx.dir/build.make:678: devilutionx] Error 1
make[1]: *** [CMakeFiles/Makefile2:77: CMakeFiles/devilutionx.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```
If I add it only to the header then GCC is ok with it but MSVC throws:
`button.obj : error LNK2001: unresolved external symbol "unsigned char const * const dvl::btnData" (?btnData@dvl@@3QBEB) [C:\projects\devilutionx\devilutionx.vcxproj]`

Reply:
OK, I'll try to fix that in a follow-up PR

Reply:
Thanks that's highly appreciated :)

I have started looking into rendering the menu directly to the framebuffer instead of having the overhead of the back buffer conversion, results look pretty good so fare.

Reply:
The cpp file was missing header include, and `const` globals have internal linkage by default. Because the header file wasn't included, the compiler didn't see that we wanted these arrays `extern`.

Reply:
Ha ha, talk about missing the forest for the trees. Thanks for fixing this.